js/apps/admin-ui/src/components/realm-selector/RealmSelector.tsx

```diff
@@ -81,43 +91,52 @@ const ContextSelectorItemLink = ({
 export const RealmSelector = () => {
   const { realm } = useRealm();
   const { realms, refresh } = useRealms();
+  const [realmDisplayName, setRealmDisplayName] = useState("");
   const { whoAmI } = useWhoAmI();
   const [open, setOpen] = useState(false);
   const [search, setSearch] = useState("");
   const { t } = useTranslation();
   const recentRealms = useRecentRealms();
 
-  const all = useMemo(
-    () =>
-      recentRealms
-        .filter((r) => r !== realm)
-        .map((name) => {
-          return { name, used: true };
-        })
-        .concat(
-          realms
-            .filter((name) => !recentRealms.includes(name) || name === realm)
-            .map((name) => ({ name, used: false })),
-        ),
-    [recentRealms, realm, realms],
-  );
+  const all = useMemo(() => {
+    const displayName = realms.find((r) => r.name === realm)?.displayName;
+    setRealmDisplayName(displayName ? displayName : "");
+    return realms
+      .filter((r) => r.name !== realm)
+      .map((realm) => {
+        const used = recentRealms.some((name) => name === realm.name);
+        return { realm, used };
+      })
+      .sort((r1, r2) => {
+        if (r1.used == r2.used) return 0;
+        if (r1.used) return -1;
+        if (r2.used) return 1;
+        return 0;
+      });
+  }, [recentRealms, realm, realms]);
 
   const filteredItems = useMemo(
     () =>
       search.trim() === ""
         ? all
-        : all.filter((r) =>
-            r.name.toLowerCase().includes(search.toLowerCase()),
+        : all.filter(
+            (r) =>
+              r.realm.name.toLowerCase().includes(search.toLowerCase()) ||
```

Comment:
Perhaps we can `toLowerCase()` the contents of `search` outside of the filter so that it doesn't do it on every iteration:

```js
const normalizedSearch = search.trim().toLowerCase();

if (normalizedSearch.length === 0) {
  return all;
}

return all.filter(({ realm }) => (
  realm.name.toLowerCase().includes(normalizedSearch) ||
  realm.displayName ?  realm.displayName.toLowerCase().includes(normalizedSearch) : false,
)
```